src/main/java/pp/springdata/todo/TodoController.java

```diff
@@ -0,0 +1,89 @@
+package pp.springdata.todo;
+
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Controller
+public class TodoController {
+
+    private final TodoRepository todoRepository;
+
+    public TodoController(TodoRepository todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model, @RequestParam(required = false) Category category,
+                       @RequestParam(required = false, defaultValue = "false") Boolean archive) {
+        List<Todo> todoList;
+        if (category != null) {
+            todoList = todoRepository.findByCategoryOrderByDeadline(category);
+        } else {
+            todoList = todoRepository.findAll(Sort.by("deadline"));
```

Comment:
Widzę że chciałaś spróbować sobie z obiektem `Sort`. Wyżej masz sortowanie w metodzie a tutaj masz z obiektem `Sort`, staraj się trzymać jednego stylu :) Ale to tak na przyszłość bo strzelam, że tutaj chciałaś sobie popróbować.